zookeeper/core/task.py

```diff
@@ -0,0 +1,58 @@
+import inspect
+
+import click
+
+from zookeeper.core.cli import ConfigParam, cli
+from zookeeper.core.component import component, configure
+from zookeeper.core.utils import convert_to_snake_case
+
+
+def task(cls):
+    """
+    A decorator which turns a class into a Zookeeper task, which is a Zookeeper
+    method with an argument-less `run` method.
+
+    Tasks are runnable through the CLI. Upon execution, the task is instantiated
+    and all component fields are configured using configuration passed as CLI
+    arguments of the form `field_name=field_value`, and then the `run` method is
+    called.
+    """
+
+    cls = component(cls)
+
+    if "run" not in dir(cls) or not (
+        inspect.isfunction(cls.run) or inspect.ismethod(cls.run)
+    ):
```

Comment:
What about:
```python
if hasattr(cls, "run") and callable(cls.run)
```

Is it because `hasattr` actually calls `getattr`?

Reply:
`hasattr` does call `getattr` but I actually don't think that's a problem here, so let's run with it. I guess technically we should probably check that `run` is not an inherited field that happens to be a function because that would be very weird behaviour.

Reply:
Actually on second thoughts we don't need to check, this is fine.